src/main/java/com/google/edith/ShelfDataReader.java

```diff
@@ -0,0 +1,102 @@
+package com.google.edith;
+
+import com.google.common.base.Splitter;
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+
+/** Processes data file of product expiration information to populate future user grocery lists. */
+public class ShelfDataReader {
+  private static final String NO_EXPIRATION = "NO_EXPIRATION";
+
+  /** Finds the specified product in the file. */
+  public static String readFile(String itemName) {
+    URL jsonresource = ShelfDataReader.class.getClassLoader().getResource("foodkeeper.json");
```

Comment:
nit: jsonResource

I think this might need to be syncd with the previous change?